### PR TITLE
Adding request and response payloads for the Sign operation

### PR DESCRIPTION
**Pull request by danetrain**
This change adds the request and response payloads for the Sign operation.

### CODE REVIEW

**Comment by danetrain** [comment]
Peter, I'm getting some weird unit test failures locally. The read() method of the SignResponsePayload works fine if I call it once in a test, but when I call it twice in one unit test, the second call fails with the error:

struct.error: unpack requires a string argument of length 4

**Comment by PeterHamilton** [comment]
It's been a while since I thought about this issue. The `BytearrayStream` object that we use for encodings is mutable. When the encoding gets read, it actually gets consumed by the reader. The bytes get pulled off the buffer and converted into whatever attribute or field is getting built at that moment, and the buffer shrinks. This makes it easy to track where we are in the buffer when reading; we're always at the beginning. When we use `is_tag_next`, we're peeking ahead into the buffer without pulling off bytes. When the reader is done reading/building itself, the buffer should be empty (that's what `is_oversized` checks for).

The simple fix is to just make a deep copy of the encoding and use that each time you want to read it in.

Long term, I've thought about changing how the `BytearrayStream` behaves (e.g., turning it into a static buffer). To be honest, it's not usually a problem. It's one of those library internal details that I just know and no one else thinks about.

**Comment by danetrain** [comment]
Ok, I will update these tests to create a new Bytearraystream for each test.

File: kmip/tests/unit/core/messages/payloads/test_sign.py [hunk 1]
```diff
@@ -0,0 +1,752 @@
...
+
+        payload = sign.SignResponsePayload()
+        args = (self.incomplete_encoding, )
+        self.assertRaisesRegexp(
+            ValueError,
+            "invalid payload missing the signature data attribute",
+            payload.read,
+            *args
+        )
+
```
Comment (PeterHamilton) [inline kmip/tests/unit/core/messages/payloads/test_sign.py]: The `read` tests look good but there need to be corresponding `write` tests as well.

**Comment by danetrain** [comment]
Added write() tests for the response payload.

File: kmip/tests/unit/core/messages/payloads/test_sign.py [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+    def test_not_equal_on_type_mismatch(self):
+        """
+        Test that the inequality operator returns True when comparing two
+        Sign request payloads with different types.
+        """
+        a = sign.SignRequestPayload()
+        b = 'invalid'
+
+        self.assertFalse(a == b)
+        self.assertFalse(b == a)
```
Comment (PeterHamilton) [inline kmip/tests/unit/core/messages/payloads/test_sign.py]: Update these two asserts to use `!=` instead.

**Pull request merged**